lib/ecto/adapters/tds/connection.ex

```diff
@@ -300,10 +300,74 @@ if Code.ensure_loaded?(Tds) do
     end
 
     @impl true
-    def explain_query(_conn, _query, _params, _opts) do
-      raise Tds.Error, "EXPLAIN is not supported by Ecto.Adapters.TDS at the moment"
+    def explain_query(conn, query, params, opts) do
+      params = prepare_params(params)
+
+      case Tds.query_multi(conn, build_explain_query(query), params, opts) do
+        {:ok, [_, %Tds.Result{} = result, _]} ->
+          {:ok, format_result_as_table(result.columns, result.rows)}
+
+        error ->
+          error
+      end
+    end
+
+    def build_explain_query(query) do
+      [
+        "SET STATISTICS XML ON; ",
+        "SET STATISTICS PROFILE ON; ",
+        query,
+        "; ",
+        "SET STATISTICS XML OFF; ",
+        "SET STATISTICS PROFILE OFF;"
+      ]
+      |> IO.iodata_to_binary()
+    end
+
+    defp format_result_as_table(columns, rows) do
```

Comment:
Those function are exactly the same as used on myxql adapter, should I move those to `Ecto.Adapters.SQL` and put a `@doc false`?

Reply:
I think we can introduce a function called `format_table` in there and make it generally available. :)